src/flash-source-buffer.js

```diff
@@ -77,18 +90,43 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
     this.updating = false;
     this.timestampOffset_ = 0;
 
-    // TS to FLV transmuxer
-    this.segmentParser_ = new flv.Transmuxer();
-    this.segmentParser_.on('data', this.receiveBuffer_.bind(this));
     encodedHeader = window.btoa(
       String.fromCharCode.apply(
         null,
         Array.prototype.slice.call(
-          this.segmentParser_.getFlvHeader()
+          flv.getFlvHeader()
         )
       )
     );
-    this.mediaSource_.swfObj.vjs_appendBuffer(encodedHeader);
+
+    // create function names with added randomness for the global callbacks flash will use
+    // to get data from javascript into the swf. Random strings are added as a safety
+    // measure for pages with multiple players since these functions will be global
+    // instead of per instance.
+    this.flashEncodedHeaderName_ = 'vjs_flashEncodedHeader_' +
+                                   this.mediaSource_.player_.id() +
+                                   generateRandomString();
+    this.flashEncodedDataName_ = 'vjs_flashEncodedData_' +
+                                   this.mediaSource_.player_.id() +
+                                   generateRandomString();
+
+    /* eslint-disable camelcase */
```

Comment:
good call, it was spotting `window.vjs_flashEncodedData_` before the randomness change happened